File: Tokens/PlayCountToken.cs
using iTunesLib;

namespace iTunesRichPresence_Rewrite.Tokens
{
    class PlayCountToken : IToken {
        public string DisplayName => "Play count";
        public string Token => "%play_count";
        public bool ShowInToolbox => true;
        public string GetText(IiTunes iTunes) {
            return iTunes.CurrentTrack.PlayedCount.ToString();
        }
    }
}

File: src/main/templating/tokens/PlayCountToken.ts
import { Token } from '../types';
import { CurrentSong } from '../../itunes/types';

export default class PlayCountToken implements Token {
    token = 'play_count';
    displayName = 'Play Count';

    render(currentSong: CurrentSong): string {
        return currentSong.PlayedCount.toString();
    }
}
